core/trino-main/src/main/java/io/trino/server/protocol/JsonEncodingUtils.java

```diff
@@ -368,8 +369,13 @@ public void encode(JsonGenerator generator, ConnectorSession session, Block bloc
             generator.writeStartObject();
             for (int i = 0; i < map.getSize(); i++) {
                 // Field name is always written as String for backward compatibility,
-                // only value is properly encoded.
-                generator.writeFieldName(mapType.getKeyType().getObjectValue(session, keyBlock, offset + i).toString());
+                // (except for VarbinaryType where we encode it as Base64 representation of its bytes)
```

Comment:
nit: key/value not (field name/value) would be better nomenclature

Reply:
it's called field name in jackson, hence the name